src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+                impulse.
+            tspan(list): Timespan from which the kernel/impulse response is
+                extracted.
+            t_bg (list): Timespan that corresponds to the background signal.
+        """
+        x_curr, y_curr = self.get_current(tspan=tspan)
+        x_pot, y_pot = self.get_potential(tspan=tspan)
+        x_sig, y_sig = self.get_signal(signal_name, tspan=tspan, t_bg=t_bg)
+
+        if signal_name == 'M32':
+            t0 = x_curr[np.argmax(y_pot > cutoff_pot)] #time of impulse
+        elif signal_name == 'M2':
+            t0 = x_curr[np.argmax(y_pot < cutoff_pot)]
+        else:
+            print('mass not found')
+
+        x_sig = x_sig - t0
+
+        y_sig = y_sig[x_sig>0]
+        x_sig = x_sig[x_sig>0]
+
+        y_curr = y_curr[x_curr>t0]
+        x_curr = x_curr[x_curr>t0]
+        y_pot = y_pot[x_pot>t0]
+        x_pot = x_pot[x_pot>t0]
+
+        kernel = Kernel(
+            MS_data = np.array([x_sig, y_sig]),
+            EC_data = np.array([x_curr, y_curr, x_pot, y_pot])
+            )
+
+
+        return kernel
+
+
+
+class Kernel:
+    """
+    Kernel class implementing different functionalities for datatreatment of
+    kernel/impulse response data.
+    """
+    def __init__ (
+        self,
+        parameters = {},
+        MS_data = None,
+        EC_data = None,
+    ):
+        """Initializes a Kernel object either in functional form by defining the
+        mass transport parameters or in the measured form by passing of EC-MS
+        data.
+
+        Args:
+            parameters (dict): Dictionary containing the mass transport
+                parameters with the following keys:
+                    D: Diffusion constant in liquid
+                    L: Working distance between electrode and gas/liq interface
+                    V: Gas sampling volume of the chip
+                    V_dot: Volumetric capillary flow
+                    kH: Dimensionless Henry volatility
+                MS_data (list): List of numpy arrays containing the MS signal
+                    data.
+                EC_data (list): List of numpy arrays containing the EC (time,
+                    current, potential).
+        """
+
+        if MS_data is not None and parameters:
+            raise Exception(
+                'Kernel can only be initialized with data OR parameters, not both'
+                )
+        if EC_data is not None and MS_data is not None:
+            print('Generating kernel from measured data')
+            self.type = 'measured'
+        elif parameters:
+            print('Generating kernel from parameters')
+            self.type = 'functional'
+        else:
+            print('Generating blank kernel')
+            self.type = None
+
+        self.params = parameters
+        self.MS_data = MS_data
+        self.EC_data = EC_data #x_curr, y_curr, x_pot, y_pot
+
+    @property
+    def sig_area(self):
```

Comment:
This method is an argument for `Kernel` inheriting from `Measurement`, as `Measurement` should have this method. A discussion point for later.